src/DeviceStatusPage/AttributesCard.tsx

```diff
@@ -78,30 +79,34 @@ const AttributesCard = ({
   onNewAttributeClick,
   onEditAttributeClick,
   onRemoveAttributeClick,
-}: AttributesCardProps): React.ReactElement => (
-  <FullHeightCard xs={12} md={6} className="mb-4">
-    <Card.Header as="h5">Attributes</Card.Header>
-    <Card.Body className="d-flex flex-column">
-      {device.attributes.size > 0 ? (
-        <AttributesTable
-          attributes={device.attributes}
-          onEditAttributeClick={onEditAttributeClick}
-          onRemoveAttributeClick={onRemoveAttributeClick}
-        />
-      ) : (
-        <p>Device has no attribute</p>
-      )}
-      <div className="mt-auto">
-        <Button
-          variant="primary"
-          onClick={onNewAttributeClick}
-          disabled={device.deletionInProgress}
-        >
-          Add attribute
-        </Button>
-      </div>
-    </Card.Body>
-  </FullHeightCard>
-);
+}: AttributesCardProps): React.ReactElement => {
+  const astarte = useAstarte();
+  return (
+    <FullHeightCard xs={12} md={6} className="mb-4">
+      <Card.Header as="h5">Attributes</Card.Header>
+      <Card.Body className="d-flex flex-column">
+        {device.attributes.size > 0 ? (
+          <AttributesTable
+            attributes={device.attributes}
+            onEditAttributeClick={onEditAttributeClick}
+            onRemoveAttributeClick={onRemoveAttributeClick}
+          />
+        ) : (
+          <p>Device has no attribute</p>
+        )}
+        <div className="mt-auto">
+          <Button
+            variant="primary"
+            onClick={onNewAttributeClick}
+            hidden={!astarte.token?.can('appEngine', 'PATCH', `/devices/${device.id}`)}
+            disabled={device.deletionInProgress}
+          >
+            Add attribute
+          </Button>
```

Comment:
This check should be run also to know whether to display the actions to edit or remove an attribute in the AttributesTable.